### PR TITLE
comrak: update to 0.18.0.

### PR DESCRIPTION
**Pull request by jcgruenhage**
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

**Comment by tranzystorek-io** [comment]
Comrak uses oniguruma, we can link void's dynlibs

**Comment by jcgruenhage** [comment]
Sorry, hadn't seen your PR, but you're right, we should link oniguruma dynamically

**Comment by jcgruenhage** [comment]
To make sure this linking is happening automatically the way we'd want it to, I've added the env var to the rust build helper instead of the comrak template. If I should split that out into a separate PR though, I could do that as well.

File: srcpkgs/comrak/template
Comment (tranzystorek-io) [inline srcpkgs/comrak/template:1]: Is the rust helper included automatically?

Comment (jcgruenhage) [reply srcpkgs/comrak/template:1]: `build_style=cargo` includes `build_helper=rust`, yea

Comment (jcgruenhage) [reply srcpkgs/comrak/template:1]: (See https://github.com/void-linux/void-packages/blob/master/common/environment/build-style/cargo.sh#L11)

**Pull request merged**